src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1287,6 +1290,20 @@ def _update_dict(dict1, dict2):
     return cp
 
 
+def subnet_role_assignment_exists(cli_ctx, scope):
+    exist = False
+    network_contributor_role_id = "4d97b98b-1d4f-4787-a291-c67834d212e7"
+
+    factory = get_auth_management_client(cli_ctx, scope)
+    assignments_client = factory.role_assignments
+
+    for i in list(assignments_client.list_for_scope(scope=scope, filter='atScope()')):
```

Comment:
yes, we need to check item in the list to find the one we need.

Reply:
I meant does `list_for_scope` already return an iterable? If so, you dont need `list()`

Reply:
I don't think so. If you searched the whole repo, we also find that `azure-cli-role` using the same syntax. BTW, I didn't find the 'list_for_scope()' function signature, which gives me hard time. If you can find that, please do let me know.

Reply:
Here is [the definition of `list_for_scope`](https://github.com/Azure/azure-sdk-for-python/blob/master/azure-mgmt-authorization/azure/mgmt/authorization/v2015_07_01/operations/role_assignments_operations.py#L674). I think it should return an iterable--could you try without `list()`?

Reply:
That's because it is coming from a sdk dependency: https://github.com/Azure/azure-sdk-for-python/blob/master/azure-mgmt-authorization/azure/mgmt/authorization/v2018_01_01_preview/operations/role_assignments_operations.py#L661-L677